geonode_mapstore_client/client/js/components/DetailsPanel/DetailsTimeSeries.jsx

```diff
@@ -0,0 +1,154 @@
+/*
+ * Copyright 2025, GeoSolutions Sas.
+ * All rights reserved.
+ *
+ * This source code is licensed under the BSD-style license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+import React, { useEffect, useState } from 'react';
+import PropTypes from "prop-types";
+import isNil from 'lodash/isNil';
+import isEmpty from 'lodash/isEmpty';
+import { Checkbox, FormGroup, ControlLabel, FormControl, HelpBlock } from 'react-bootstrap';
+import Select from 'react-select';
+import Message from '@mapstore/framework/components/I18N/Message';
+import HTML from '@mapstore/framework/components/I18N/HTML';
+import { TIME_SERIES_PROPERTIES, TIME_ATTRIBUTE_TYPES, TIME_PRECISION_STEPS } from '@js/utils/ResourceUtils';
+import { getMessageById } from '@mapstore/framework/utils/LocaleUtils';
+import InfoPopover from '@mapstore/framework/components/widgets/widget/InfoPopover';
+
+const TimeSeriesSettings = ({ resource, onChange }, context) => {
+    const timeAttributes = (resource?.attribute_set ?? [])
+        .filter((attribute) => TIME_ATTRIBUTE_TYPES.includes(attribute.attribute_type))
+        .map((attribute)=> ({value: attribute.pk, label: attribute.attribute}));
+
+    if (isEmpty(timeAttributes)) return null;
+
+    const [timeseries, setTimeSeries] = useState();
+    const [error, setError] = useState();
+    useEffect(() => {
+        resource.timeseries && setTimeSeries(resource.timeseries);
+    }, [resource?.timeseries]);
+
+
+    const onChangeTimeSettings = (key, value) => {
+        const _timeseries = {
+            ...timeseries,
+            [key]: value,
+            ...(key === "presentation"
+                ? value === "LIST"
+                    // reset precision field when presentation is LIST
+                    ? {precision_value: undefined, precision_step: undefined}
+                    : {precision_value: null, precision_step: "seconds"} : undefined
+            ),
+            ...(key === "has_time"
+                ? !value
+                    // reset all time series properties when has_time is `false`
+                    ? TIME_SERIES_PROPERTIES.reduce((obj, prop) => ({...obj, [prop]: undefined}), {})
+                    : { presentation: "LIST"} : undefined
+            )
+        };
+        setTimeSeries(_timeseries);
+        setError(_timeseries.has_time ? isNil(_timeseries.attribute) && isNil(_timeseries.end_attribute) : false );
+    };
+
+    useEffect(() => {
+        if (!error && !isEmpty(timeseries)) {
+            // update resource when timeseries is changed and valid
+            onChange({timeseries}, "timeseries");
+        }
+    }, [error, JSON.stringify(timeseries)]);
```

Comment:
Why this is not done inside the onChangeTimeSettings callback?
It seems the setTimeSeries and setError state are applied only inside onChangeTimeSettings.